server/controllers/notifications.js

```diff
@@ -0,0 +1,48 @@
+const asyncHandler = require('express-async-handler');
+const Notifications = require('../models/Notifications');
+
+exports.getAllNotifications = asyncHandler(async(req, res) => {
+    const userId = req.params.id;
```

Comment:
You can get the user id through the request object rather than through a parameter and update the route accordingly.
Eg: `req.user.id`